README.md
# KCMS-Media
Kubee CMS Media Library Component ...


changelog.txt
Jul 23, 2020 - Version 4.0.1
- Fixed: Auto import folders from old version

Jul 22, 2020 - Version 4.0
- Added: New designs
- Added: Use new technology
- Added: Smooth performance for unlimited folders and large numbers of files
- Added: REST API for developers
- Added: Sort files by Name, Date, Modified, Author
- Added: Bulk select folders
- Added: Bulk delete folders
- Added: Wipe old data
- Added: Clean all data
- Added: Import from Media Library Assistant
- Added: Import from WP Media Folder by JoomUnited
- Added: Support Brizy builder
- Added: Support Fusion builder
- Added: Compatible with KnowherePro theme
- Improved: Switchers
- Improved: Text and Popups
- Improved: Documentation
- Fixed: Special characters in folder name
- Fixed: Auto detect site width
- Fixed: SQL syntax errors
- Fixed: Bugs in list view
- Fixed: Duplicate queries
- Fixed: Bugs with WPML
- Fixed: Small bugs

Mar 6, 2020 - Version 3.9
- Added: Support Thrive Architect
- Added: Language files for Taiwan 
- Fixed: Some bugs

Mar 2, 2020 – Version 3.8.2
- Fixed: List view order item bug 
- Fixed: WPML bug

Feb 23, 2020 – Version 3.8.1
- Fixed: Scroll bar
- Fixed: Some small bugs

Feb 23, 2020 – Version 3.8
- Added: New FileBird layout
- Added: New icon
- Added: Separate Folder(s) for different user
- Added: Import files from 2 media library plugins (Enhanced Media Library by webbistro and WordPress Media Library Folders by Max Foundry)
- Added: Folder sort, folder search
- Fixed: Support PolyLang
- Fixed: Count
- Fixed: Splitter
- Fixed: Rename folder in list view mode
- Fixed: Some bugs in right to left screen
- Fixed: Move large folders
- Fixed: Filter is hidden when using Smush plugin
- Removed: Unused css
- Removed: Unused libraries

Nov 11, 2019 – Version 3.5
- Added: FileBird Gallery block for Gutenberg
- Fixed: Error of updating media image size in unsaved post
- Fixed: Menu CSS
- Fixed: Remove trigger which conflicts with Divi builder
- Fixed: Conflict with Gridd Theme
- Fixed: Small bugs

Oct 28, 2019 - Version 3.4
- Removed: Unused CSS
- Fixed: Elementor bug
- Fixed: List view not load image
- Fixed: Count items in Uncategorized folder
- Fixed: Conflict CSS with ACF plugin
- Fixed: Small bugs

Oct 22, 2019 - Version 3.3
- Added: Option turn on/off load FileBird JS at front-end for page builders
- Improved: Removed CDN for customScrollBar
- Fixed: Small bug will Elementor
- Fixed: Select option doesn’t not change in modal popup upload
- Fixed: Only call move image when moving folder position

Sep 18, 2019 – Version 3.2.2
- Fixed: Sorting error in list view
- Improved: Optimize Code and Security requirements

Sep 16, 2019 – Version 3.2.1
- Fixed: Conflict with WordPress Customize Function
- Improved: Optimize Code
- Fixed: Some small bugs

Sep 13, 2019 – Version 3.2
- Fixed: "nt_wmc_folder" in Google Search Console results
- Fixed: Error in full front end Divi Page Builder editor
- Fixed: PHP notice on line 582 and 586
- Fixed: Load JS script on front end
- Fixed: Error "FileBird is loading" on Elementor builder
- Fixed: Compatible with FooGallery and Final Tiles Grid Gallery
- Fixed: Warning notice in WordPress
- Fixed: Margin Overlay
- Fixed: Other small bugs
- Improved: Performance and Optimize Code

Jul 29, 2019 - Version 3.1.2
- Improved: Fast append treeview
- Improved: Prevent public folders link
- Improved: CSS UI
- Fixed: Prevent public taxonomy folder
- Fixed: Query database callback all time
- Fixed: FileBird treeview show on the top left in product editor in WooCommerce
- Fixed: Can’t drop file when creating the first folder
- Fixed: Work with WPML when Post Types Translation: Media tick not translatable
- Fixed: Bulk select moving images into folders by itself
- Fixed: Move folder with length folder text
- Fixed: Resize bar not save in minimum width
- Fixed: Remove map file library
- Fixed: Conflict with photo gallery

Jun 10, 2019 - Version 3.1.1
- Added: Support RTL languages
- Added: Support Advanced Custom Field
- Improved: Reduce plugin size
- Fixed: Small bugs

Jun 04, 2019 - Version 3.1
- Improved: Change drag-drop library
- Improved: Optimize drag-drop
- Improved: Change drag-drop to trigger folders, list view
- Improved: Change syntax jQuery
- Fixed: CSS not show in all folders treeview from classic editor
- Fixed: Move many files
- Fixed: Move all categories
- Fixed: Auto Deselect bulk when none file
- Fixed: Count file selected in bulk select
- Fixed: Conflict with Avada theme
- Updated: CSS
- Removed: Unused files

May 29, 2019 - Version 3.0.1
- Fixed: Move files from Uncategorized and All files folders
- Fixed: Count files when moved

May 28, 2019 - Version 3.0
- Added: Treeview interface in page, post, custom post type
- Added: Treeview interface in Divi Builder, Beaver Builder, Visual Composer, WPBakery Page Builder, Elementor Builder
- Added: Treeview interface for Gutenberg
- Added: Bulk auto deselect when change folder
- Added: Compatible with WPML 4.x version
- Fixed: WPML 3.9 conflict
- Fixed: Move one file not click image
- Fixed: CSS over when edit folder 
- Fixed: Load too much database
- Fixed: Treeview interface Elementor Builder
- Fixed: Upload images
- Improved: No load FileBird files on frontend if not use builder
- Improved: Optimize FileBird
- Improved: UI/UX
- Removed: Unused files

Mar 13, 2019 - Version 2.7
- Added: Refresh folder
- Fixed: Categories dropdown from Media Library popup
- Fixed: Some small bugs

Mar 4, 2019 - Version 2.6.1
- Updated: Languages and Documentation
- Fixed: Press double clicks for OK/cancel button

Feb 27, 2019 - Version 2.6
- Fixed: Scroll folder tree
- Fixed: Bug for Elementor
- Fixed: Languages
- Fixed: Removed WordPress notification
- Improved: Doesn't jump to destination folder when move files

Feb 14, 2019 - Version 2.5

views/pages/html-settings.php
<?php
defined('ABSPATH') || exit;

$current_tab = ((isset($_GET['tab']) && in_array(sanitize_text_field($_GET['tab']), array('settings', 'update-db', 'update', 'api'))) ? sanitize_text_field($_GET['tab']) : 'settings');

$countEnhancedFolder = count($helpers::foldersFromEnhanced(0, true));
$countWpmlfFolder = count($helpers::foldersFromWpmlf(0, true));
$countWpmfFolder = count($helpers::foldersFromWpmf(0, true));
$countRealMediaFolder = count($helpers::foldersFromRealMedia(-1, true));

?>
<div class="">
  
  <form action="options.php" method="POST" id="post" autocomplete="off">
    <?php settings_fields('njt_fbv'); ?>
    <?php do_settings_sections('njt_fbv'); ?>
    <nav class="nav-tab-wrapper">
        <a href="<?php echo add_query_arg(array('page' => 'filebird-settings', 'tab' => 'settings'), admin_url('options-general.php')); ?>" class="nav-tab <?php echo $current_tab == 'settings' ? 'nav-tab-active' : '' ?>"><?php _e('Settings', 'filebird') ?></a>
        <a href="<?php echo add_query_arg(array('page' => 'filebird-settings', 'tab' => 'update-db'), admin_url('options-general.php')); ?>" class="nav-tab <?php echo $current_tab == 'update-db' ? 'nav-tab-active' : '' ?>"><?php _e('Update Database', 'filebird') ?></a>
        
        <?php if(($countEnhancedFolder + $countWpmlfFolder + $countWpmfFolder + $countRealMediaFolder) > 0) : ?>
          <a href="<?php echo add_query_arg(array('page' => 'filebird-settings', 'tab' => 'update'), admin_url('options-general.php')); ?>" class="nav-tab <?php echo $current_tab == 'update' ? 'nav-tab-active' : '' ?>"><?php _e('Import', 'filebird') ?></a>
        <?php endif; ?>

        <a href="<?php echo add_query_arg(array('page' => 'filebird-settings', 'tab' => 'api'), admin_url('options-general.php')); ?>" class="nav-tab <?php echo $current_tab == 'api' ? 'nav-tab-active' : '' ?>"><?php _e('API', 'filebird') ?></a>
    </nav>
    <?php if($current_tab == 'settings') : ?>
    <h1><?php _e('Settings', 'filebird'); ?></h1>
    <table class="form-table">
      <tr>
        <th scope="row" style="width: 30%;">
          <label for="njt_fbv_folder_per_user"><?php _e('Each user has his own folders?', 'filebird'); ?></label>
        </th>
        <td>
          <label class="njt-switch">
            <input type="checkbox" name="njt_fbv_folder_per_user" class="njt-submittable" id="njt_fbv_folder_per_user"  value="1" <?php checked(get_option('njt_fbv_folder_per_user'), '1'); ?> />
            <span class="slider round"></span>
          </label>
        </td>
      </tr>
      <?php
      /**
       <tr>
        <th scope="row">
          <label><?php _e('Wipe old data', 'filebird'); ?></label>
          <p class="description" style="font-weight: 400"><?php _e('This action will delete FileBird data in version 3.0 and earlier installs.', 'filebird'); ?></p>
        </th>
        <td>
        <button type="button" class="button button-primary njt_fbv_wipe_old_data"><?php _e('Wipe', 'filebird'); ?></button>
        </td>
      </tr>
       */
      ?>
      <tr>
        <th scope="row">
          <label><?php _e('Clear all data', 'filebird'); ?></label>
          <p class="description" style="font-weight: 400"><?php _e('This action will delete all FileBird data, FileBird settings and bring you back to WordPress default media library.', 'filebird'); ?></p>
        </th>
        <td>
        <button type="button" class="button button-primary njt_fbv_clear_all_data"><?php _e('Clear', 'filebird'); ?></button>
        </td>
      </tr>
    </table>
    <?php //submit_button(); ?>
    <?php elseif ($current_tab == 'update-db'): ?>
    <h1><?php _e('Update Database', 'filebird'); ?></h1>
    <table class="form-table njt-fb-import-tbl">
        <tbody>
            <tr>
                <th scope="row">
                <label><?php _e('Import from old version', 'filebird'); ?></label>
                <p class="description" style="font-weight: 400"><?php _e('By running this action, all folders created in version 3.9 & earlier installs will be imported.', 'filebird'); ?></p>
                </th>
                <td>
                <button type="button" class="button button-primary njt_fbv_import_from_old_now"><?php _e('Update now', 'filebird'); ?></button>
                </td>
            </tr>
        </tbody>
    </table>

    <?php elseif ($current_tab == 'update'): ?>
      <h1><?php _e('Import to FileBird', 'filebird'); ?></h1>
      <p style="margin-top:0;">
        <?php _e('Import categories/folders from other plugins. We import virtual folders, your website will be safe, don\'t worry ;)', 'filebird') ?>
      </p>
        <table class="form-table njt-fb-import-tbl">
            <tbody>
                <tr class="<?php echo $countEnhancedFolder <= 3 ? 'hidden' : ''; ?>">
                    <th scope="row">
                      <label for="">
                        <?php echo __('Enhanced Media Library plugin by wpUXsolutions', 'filebird') ?>
                      </label>
                      <p class="description" style="font-weight: 400">
                        <?php
                          $str = __('We found you have <strong>(%1$s)</strong> categories you created from <strong>Enhanced Media Library</strong> plugin.', 'filebird');
                          if($countEnhancedFolder > 0) {
                            $str .= __(' Would you like to import to <strong>FileBird</strong>?', 'filebird');
                          }
                          echo (sprintf($str, $countEnhancedFolder));
                        ?>
                      </p>
                    </th>
                    <td>
                      <div class="fbv-btn-wrapper-import">
                        <?php if($countEnhancedFolder > 0) : ?>
                          <button class="button button-primary button-large njt-fb-import" data-site="enhanced" type="button" data-count="<?php echo $countEnhancedFolder; ?>"><?php _e('Import Now', 'filebird') ?></button>
                        <?php endif; ?>
                      </div>
                    </td>
                </tr>
                <tr class="fbv-row-breakline <?php echo $countEnhancedFolder <= 3 ? 'hidden' : ''; ?>">
                  <td>
                    <span class="fbv-breakline"></span>
                  </td>
                  <td>
                    <span class="fbv-breakline"></span>
                  </td>
                </tr>
                <tr class="<?php echo $countWpmlfFolder <= 3 ? 'hidden' : ''; ?>">
                    <th scope="row">
                      <label for="">
                        <?php echo __('WordPress Media Library Folders by Max Foundry', 'filebird') ?>
                      </label>
                      <p class="description" style="font-weight: 400">
                        <?php
                          $str = __('We found you have <strong>(%1$s)</strong> categories you created from <strong>WordPress Media Library Folders</strong> plugin.', 'filebird');
                          if($countWpmlfFolder > 0) {
                            $str .= __(' Would you like to import to <strong>FileBird</strong>?', 'filebird');
                          }
                          echo (sprintf($str, $countWpmlfFolder));
                        ?>
                      </p>
                    </th>
                    <td>
                      <div class="fbv-btn-wrapper-import">
                        <?php if($countWpmlfFolder > 0) : ?>
                          <button class="button button-primary button-large njt-fb-import" data-site="wpmlf" type="button" data-count="<?php echo $countWpmlfFolder; ?>"><?php _e('Import Now', 'filebird') ?></button>
                          <?php endif; ?>
                      </div>
                    </td>
                </tr>
                <tr class="fbv-row-breakline <?php echo $countWpmlfFolder <= 3 ? 'hidden' : ''; ?>">
                  <td>
                    <span class="fbv-breakline"></span>
                  </td>
                  <td>
                    <span class="fbv-breakline"></span>
                  </td>
                </tr>
                <tr class="<?php echo $countWpmfFolder <= 3 ? 'hidden' : ''; ?>">
                    <th scope="row">
                      <label for="">
                        <?php echo __('WP Media folder by Joomunited', 'filebird') ?>
                      </label>
                      <p class="description" style="font-weight: 400">
                        <?php
                          $str = __('We found you have <strong>(%1$s)</strong> categories you created from <strong>WP Media folder</strong> plugin.', 'filebird');
                          if($countWpmfFolder > 0) {
                            $str .= __(' Would you like to import to <strong>FileBird</strong>?', 'filebird');
                          }
                          echo (sprintf($str, $countWpmfFolder));
                        ?>
                      </p>
                    </th>
                    <td>
                      <div class="fbv-btn-wrapper-import">
                        <?php if($countWpmfFolder > 0) : ?>
                          <button class="button button-primary button-large njt-fb-import" data-site="wpmf" type="button" data-count="<?php echo $countWpmfFolder; ?>"><?php _e('Import Now', 'filebird') ?></button>
                          <?php endif; ?>
                      </div>
                    </td>
                </tr>
                <tr class="fbv-row-breakline <?php echo $countWpmfFolder <= 3 ? 'hidden' : ''; ?>">
                  <td>
                    <span class="fbv-breakline"></span>
                  </td>
                  <td>
                    <span class="fbv-breakline"></span>
                  </td>
                </tr>
                <tr class="<?php echo $countRealMediaFolder <= 3 ? 'hidden' : ''; ?>">
                    <th scope="row">
                      <label for="">
                        <?php echo __('WP Real Media Library by devowl.io GmbH', 'filebird') ?>
                      </label>
                      <p class="description" style="font-weight: 400">
                        <?php
                          $str = __('We found you have <strong>(%1$s)</strong> categories you created from <strong>WP Real Media Library</strong> plugin.', 'filebird');
                          if($countRealMediaFolder > 0) {
                            $str .= __(' Would you like to import to <strong>FileBird</strong>?', 'filebird');
                          }
                          echo (sprintf($str, $countRealMediaFolder));
                        ?>
                      </p>
                    </th>
                    <td>
                      <div class="fbv-btn-wrapper-import">
                        <?php if($countRealMediaFolder > 0) : ?>
                          <button class="button button-primary button-large njt-fb-import" data-site="realmedia" type="button" data-count="<?php echo $countRealMediaFolder; ?>"><?php _e('Import Now', 'filebird') ?></button>
                          <?php endif; ?>
                      </div>
                    </td>
                </tr>
                <tr class="fbv-row-breakline <?php echo $countRealMediaFolder <= 3 ? 'hidden' : ''; ?>">
                  <td>
                    <span class="fbv-breakline"></span>
                  </td>
                  <td>
                    <span class="fbv-breakline"></span>
                  </td>
                </tr>
            </tbody>
        </table>
    <?php elseif ($current_tab == 'api'): ?>
    <h1><?php _e('REST API', 'filebird'); ?></h1>
    <?php _e("An API to run Get folders & Set attachments", 'filebird') ?><br/>
    <?php echo __('Please see FileBird API for developers <a target="_blank" href="https://ninjateam.gitbook.io/filebird/api">here</a>.', 'filebird') ?>
    <table class="form-table">
      <tbody>
        <tr>
            <th scope="row">
              <label for="">
                <?php echo __('API key', 'filebird') ?>
              </label>
            </th>
            <td>
              <?php
              $key = get_option('fbv_rest_api_key', '');
              $classes = array('regular-text');
              if(strlen($key) == 0) {
                $classes[] = 'hidden';
              }
              $classes = array_map('esc_attr', $classes);
              ?>
              <input type="text" id="fbv_rest_api_key" class="<?php echo implode(' ', $classes); ?>" value="<?php echo esc_attr($key); ?>" onclick="this.select()" />
              <button type="button" class="button button-primary fbv_generate_api_key_now"><?php _e('Generate'); ?></button>
            </td>
        </tr>
      </tbody>
    </table>
    <?php endif; ?>
  </form>
</div>
<?php
if(isset($_GET['autorun']) && ($_GET['autorun'] == 'true')) {
  ?>
  <script>
    var njt_auto_run_import = true;
    var njt_fb_settings_page = '<?php echo add_query_arg(array('page' => 'filebird-settings', 'tab' => 'update-db'), admin_url('options-general.php')); ?>';
    jQuery(document).ready(function($){
      jQuery('.njt_fbv_import_from_old_now').click();
    })
  </script>
  <?php
}
?>